src/main/java/software/bernie/techarium/tile/base/FunctionalTileBase.java

```diff
@@ -0,0 +1,116 @@
+package software.bernie.techarium.tile.base;
+
+import net.minecraft.block.BlockState;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.entity.player.PlayerInventory;
+import net.minecraft.entity.player.ServerPlayerEntity;
+import net.minecraft.inventory.container.Container;
+import net.minecraft.inventory.container.INamedContainerProvider;
+import net.minecraft.nbt.CompoundNBT;
+import net.minecraft.network.NetworkManager;
+import net.minecraft.network.play.server.SUpdateTileEntityPacket;
+import net.minecraft.tileentity.ITickableTileEntity;
+import net.minecraft.tileentity.TileEntityType;
+import net.minecraft.util.ActionResultType;
+import net.minecraft.util.Direction;
+import net.minecraft.util.math.AxisAlignedBB;
+import net.minecraft.util.text.ITextComponent;
+import net.minecraft.util.text.TextFormatting;
+import net.minecraft.util.text.TranslationTextComponent;
+import net.minecraftforge.common.capabilities.Capability;
+import net.minecraftforge.common.util.LazyOptional;
+import net.minecraftforge.energy.CapabilityEnergy;
+import net.minecraftforge.fluids.capability.CapabilityFluidHandler;
+import net.minecraftforge.fml.network.NetworkHooks;
+import net.minecraftforge.items.CapabilityItemHandler;
+import software.bernie.techarium.display.container.AutomaticContainer;
+import software.bernie.techarium.machine.addon.fluid.FluidTankAddon;
+import software.bernie.techarium.machine.addon.inventory.InventoryAddon;
+import software.bernie.techarium.machine.controller.MachineController;
+import software.bernie.techarium.machine.interfaces.recipe.IForcedRecipe;
+import software.bernie.techarium.machine.interfaces.recipe.IMachineRecipe;
+import software.bernie.techarium.machine.interfaces.recipe.IRecipeMachine;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+import static software.bernie.techarium.util.StaticHandler.*;
+
+public abstract class FunctionalTileBase extends MachineTileBase implements ITickableTileEntity {
+    public FunctionalTileBase(TileEntityType<?> tileEntityTypeIn) {
```

Comment:
Idk what the point of this is but we should try to avoid creating big inheritance trees and refactor stuff to use Traits/Behaviours instead.